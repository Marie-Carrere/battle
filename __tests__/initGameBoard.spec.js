import { initGameBoard } from '../scripts/initGameBoard'

describe('initialize the gameboard', () => {
    it('should initialized and create the board with 5 rows and 5 columns into the DOM', function() {
        document.body.innerHTML = '<table id="gameboard"></table>';

        let gameboard = [
            [0,0,0,0,0],
            [0,0,0,0,0],
            [0,0,0,0,0],
            [0,0,0,0,0],
            [0,0,0,0,0]
        ]
        initGameBoard(gameboard);

        expect(document.getElementById('gameboard')).toMatchSnapshot();
    })
})