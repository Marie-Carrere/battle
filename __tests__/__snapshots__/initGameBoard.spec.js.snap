// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialize the gameboard should initialized and create the board with 5 rows and 5 columns into the DOM 1`] = `
<table
  id="gameboard"
>
  <tr
    id="0"
  >
    <td
      class="0"
    />
    <td
      class="1"
    />
    <td
      class="2"
    />
    <td
      class="3"
    />
    <td
      class="4"
    />
  </tr>
  <tr
    id="1"
  >
    <td
      class="0"
    />
    <td
      class="1"
    />
    <td
      class="2"
    />
    <td
      class="3"
    />
    <td
      class="4"
    />
  </tr>
  <tr
    id="2"
  >
    <td
      class="0"
    />
    <td
      class="1"
    />
    <td
      class="2"
    />
    <td
      class="3"
    />
    <td
      class="4"
    />
  </tr>
  <tr
    id="3"
  >
    <td
      class="0"
    />
    <td
      class="1"
    />
    <td
      class="2"
    />
    <td
      class="3"
    />
    <td
      class="4"
    />
  </tr>
  <tr
    id="4"
  >
    <td
      class="0"
    />
    <td
      class="1"
    />
    <td
      class="2"
    />
    <td
      class="3"
    />
    <td
      class="4"
    />
  </tr>
</table>
`;
